python/runtime/utils/PyRestRemoteClient.cpp

```diff
@@ -16,6 +16,407 @@ class PyRestRemoteClient : public cudaq::BaseRemoteRestRuntimeClient {
+        // This is a large response that needs to be downloaded
+        const std::string downloadUrl = resultJs["responseReference"];
+        const std::string reqId = resultJs["reqId"];
+        cudaq::info("Download result for Request Id {} at {}", reqId,
+                    downloadUrl);
+        llvm::SmallString<32> tempDir;
+        llvm::sys::path::system_temp_directory(/*ErasedOnReboot*/ true,
+                                               tempDir);
+        std::filesystem::path resultFilePath =
+            std::filesystem::path(tempDir.c_str()) / (reqId + ".zip");
+        m_restClient.download(downloadUrl, resultFilePath.string(),
+                              /*enableLogging=*/false, /*enableSsl=*/true);
+        cudaq::info("Downloaded zip file {}", resultFilePath.string());
+        std::filesystem::path unzipDir =
+            std::filesystem::path(tempDir.c_str()) / reqId;
+        // Unzip the response
+        cudaq::utils::unzip(resultFilePath, unzipDir);
+        std::filesystem::path resultJsonFile =
+            unzipDir / (reqId + "_result.json");
+        if (!std::filesystem::exists(resultJsonFile)) {
+          if (optionalErrorMsg)
+            *optionalErrorMsg =
+                "Unexpected response file: missing the result JSON file.";
+          return false;
+        }
+        std::ifstream t(resultJsonFile.string());
+        std::string resultJsonFromFile((std::istreambuf_iterator<char>(t)),
+                                       std::istreambuf_iterator<char>());
+        try {
+          resultJs["response"] = json::parse(resultJsonFromFile);
+        } catch (...) {
+          if (optionalErrorMsg)
+            *optionalErrorMsg =
+                fmt::format("Failed to parse the response JSON from file '{}'.",
+                            resultJsonFile.string());
+          return false;
+        }
+        cudaq::info(
+            "Delete response zip file {} and its inflated contents in {}",
+            resultFilePath.c_str(), unzipDir.c_str());
+        std::filesystem::remove(resultFilePath);
+        std::filesystem::remove_all(unzipDir);
+      }
+
+      if (!resultJs.contains("response")) {
+        if (optionalErrorMsg)
+          *optionalErrorMsg = "Unexpected response from the NVQC invocation. "
+                              "Missing the 'response' field.";
+        return false;
+      }
+      if (!resultJs["response"].contains("executionContext")) {
+        if (optionalErrorMsg) {
+          if (resultJs["response"].contains("errorMessage")) {
+            *optionalErrorMsg = fmt::format(
+                "NVQC failed to handle request. Server error: {}",
+                resultJs["response"]["errorMessage"].get<std::string>());
+          } else {
+            *optionalErrorMsg =
+                "Unexpected response from the NVQC response. "
+                "Missing the required field 'executionContext'.";
+          }
+        }
+        return false;
+      }
```

Comment:
There is some additional logging that we've added in `main`.

Reply:
Addressed in cf68853